Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/get.go

```diff
@@ -289,9 +291,29 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 	// use the default printer for each object
 	printer = nil
 	var lastMapping *meta.RESTMapping
+	var withKind bool = allKinds
 	w := kubectl.GetNewTabWriter(out)
 	defer w.Flush()
 
+	// determine if printing multiple kinds of
+	// objects and enforce "show-kinds" flag if so
+	for ix := range objs {
+		var mapping *meta.RESTMapping
+		if sorter != nil {
+			mapping = infos[sorter.OriginalPosition(ix)].Mapping
+		} else {
+			mapping = infos[ix].Mapping
+		}
+
+		// display "kind" column only if we have mixed resources
+		if lastMapping != nil && mapping.Resource != lastMapping.Resource {
+			withKind = true
+		}
+		lastMapping = mapping
+	}
+
+	lastMapping = nil
```

Comment:
@rhcarvalho Thanks for the feedback! I will implement this in the next update to this PR